ktor-shared/ktor-serialization/ktor-serialization-gson/jvm/src/GsonConverter.kt

```diff
@@ -8,18 +8,34 @@ import com.google.gson.*
 import io.ktor.http.*
 import io.ktor.http.content.*
 import io.ktor.serialization.*
+import io.ktor.util.*
 import io.ktor.util.reflect.*
 import io.ktor.utils.io.*
 import io.ktor.utils.io.charsets.*
 import io.ktor.utils.io.jvm.javaio.*
 import kotlinx.coroutines.*
+import kotlinx.coroutines.flow.*
+import java.io.*
 import kotlin.reflect.*
 import kotlin.reflect.jvm.*
 
 /**
- * A GSON converter for the [ContentNegotiation] plugin.
+ * A content converter that uses [Gson]
+ *
+ * @param gson a configured instance of [Gson]
+ * @param streamRequestBody if set to true, will stream request body, without keeping it whole in memory.
+ * This will set `Transfer-Encoding: chunked` header.
  */
-public class GsonConverter(private val gson: Gson = Gson()) : ContentConverter {
+public class GsonConverter(
+    private val gson: Gson = Gson(),
+    private val streamRequestBody: Boolean = true
```

Comment:
I have multiple concerns about this:
1. Changing default behaviour
2. This is app level config, applied to all responses, but is useful only for responses that return flow.

Reply:
I add a new behavior based on OutputStreamContent for all types, not only Flow, in GsonConverter and in KotlinxSerializationJsonJvmConverter (that previously used only a TextContent without streaming).
So I adopted the exact same solution to allow the user to bypass streaming (`Transfer-Encoding: chunked`) : this new `streamRequestBody` parameter to mimic the recent change you made in JacksonConverter ([here](https://github.com/ktorio/ktor/commit/3595ddff009720f3e00a59fe59897839c75fbc12)).

Reply:
> we should not change behaviour, unless it is a bug

The chunked response was not available at all with current Ktor + Gson or kotlinx-coroutines, I thought that changing the default behavior of Gson and kotlinx-serialization could be a nice improvement by making Jackson, Gson and kotlinx-serialization now all 3 consistent about chunked response inside Ktor.
I don't think a future user would understand why default chunked response is `true` with Jackson and `false` in Gson and kotlinx-serialization in Ktor, now that all 3 allow it.
But this would require a proper changelog to document this change, an this should only be included in let's say the upcoming 2.2.0 release, maybe not a minor bugfix release.

This is my opinion, but I will of course respect your decision and turn to `false` this param if you ask me to preserve the default behavior on Gson and kotlinx-serialization is better !

> Actually, why you need this flag completely here?

I think if Jackson provides this flag to turn off chunked response, then Gson and kotlinx-serialization have to provide it as well, that's why I added it in my last commit.

> What is the flag set to false and user respond with Flow?

Nice catch, I forgot to handle this case for Gson !
I can fix it and add a new unit test for chunked flag to `false` + Flow for Gson, Jackson and kotlinx-serialization.

Reply:
I prefer to keep defaults. There are way more people who bump versions and expect the same behavior than people who switch serialization libraries and expect the same behavior =)

Jackson converter has this flag for historical reasons, and other converters do not need to have it. So far, there have been no requests for it. What do you think about not having this flag at all and just use `chunked` where user respond with `Flow` and keep the rest of the behaviour as it is?

Reply:
I understand and respect this choice, so I will remove chunked support (= the `streamRequestBody` parameter) for Gson and kotlinx-serialization, and as you suggest keep `chunked` only for the `Flow` scenario.

But from my external point of view, as a recent Ktor user, I don't see why in Ktor Jackson would be the only framework to provide `streamRequestBody` option whereas Gson and kotlinx-serialization can support it too.
For Gson and kotlinx-serialization we could just set the `streamRequestBody` parameter with a `false` default value to maintain backward behaviour.

Reply:
We can do it, of course, but let's keep it a separate task. This PR is already big and complex enough.